controllers/group_controller.go

```diff
@@ -159,3 +159,36 @@ func GetUsersOfGroups(ctx *gin.Context) {
 	}
 	SendResponse(ctx, 200, groupUsers, "Fetched group users", nil)
 }
+
+func GetUsersWithoutGroupAccess(ctx *gin.Context) {
+	userID, err := utils.FetchUserIDFromCtx(ctx)
+	if err != nil {
+		SendResponse(ctx, 401, nil, "Unauthorized", errors.New("unauthorized"))
+		return
+	}
+
+	groupIDStr := ctx.Param("groupId")
+	groupID, err := uuid.Parse(groupIDStr)
+	if err != nil {
+		SendResponse(ctx, 400, nil, "Invalid group id", nil)
+		return
+	}
+	// Check user is authorized to see members of the group
+	isMember, err := service.CheckUserMemberOfGroup(ctx, userID, groupID)
```

Comment:
Move this check to the service layer. Also that comment does not make much sense